src/file.cr

```diff
@@ -355,6 +364,11 @@ class File < IO::FileDescriptor
     (stat.st_mode & LibC::S_IFMT) == LibC::S_IFLNK
   end
 
+   # Opens the file named by filename. If a file is being created, its initial
+   # permissions may be set using the perm parameter.
+   # If the optional code block is given, it will be passed the opened file as
```

Comment:
No need to talk about the optional block. Methods with blocks are different overloads, so this should be documented in the method that yields.